quiz-backend/src/Quiz.cpp

```diff
@@ -1,45 +1,66 @@
-#include "Quiz.h"
-#include "User.h"
+#include "quiz.h"
+#include "user.h"
 #include <utility>
 #include "Question.h"
 
-void Quiz::setResult(double score) {
-    result_ = score;
-}
-
-double Quiz::getResult() {
-    return result_;
-}
-
-void Quiz::generateQuestions(std::map<std::string, int> profs) {
-    // First step is to sum all of the proficiency scores to calculate the percentages
+Quiz::Quiz(std::vector<Topic>& topics) {
+    // First step is to sum all of the proficiency scores for active topics
     int sum = 0;
-    for (const auto & [key, value] : profs) {
-        sum += value;
+    for (Topic t : topics) {
+        if (t.isActive()) {
+            sum += t.getProficiency();
+        }
     }
     // Passing each percentage interval into a vector with the topic name linked to the percentage
     // If there were 5 topics with the same proficiency, we would have a vector of: [0.20, 0.40, 0.60, 0.80, 1.00]
-    std::vector<std::pair<std::string, double>> percents;
+    std::vector<std::pair<std::Topic, double>> percents;
     double nextVal = 0.0;
-    for (const auto & [key, value] : profs) {
-        std::pair<std::string, double> topicOdds;
-        topicOdds.first = key;
-        topicOdds.second = (nextVal + (20 - value) / sum);
-        percents.push_back(topicOdds); // maybe change if we dont want 0% chance for mastered topic
-        nextVal++;
+    for (Topic t : topics) {
+        std::pair<Topic, double> topicOdds;
+        if (t.isActive()) {
+            topicOdds.first = t;
+            topicOdds.second = (nextVal + (20 - t.getProficiency()) / sum);
```

Comment:
A little bit confused on where 20 comes from here, I think a named int variable would be useful.